app/controllers/samples_controller.rb

```diff
@@ -401,6 +401,7 @@ def create
     end
 
     params[:input_files_attributes] = params[:input_files_attributes].reject { |f| f["source"] == '' }
+    params[:input_files_attributes] = params[:input_files_attributes].reject { |f| f["source"].include?(SAMPLES_BUCKET_NAME) } unless current_user && current_user.role == 1
```

Comment:
what does `current_user.role == 1` mean?

Reply:
oh admin?

Reply:
Correct

Reply:
can we just do current_user.admin?  
this makes only admin can upload?

Reply:
Actually, I think we should do 

 { |f| f["source"] == '' && current_user.can_upload( f["source"]) } 

and implement the can_upload in the user model to make it more modular and generic. 
Also, current_user should always be set in this case.

Reply:
ok!